sporkfish/endgame_tablebase.py

```diff
@@ -110,9 +119,106 @@ def _should_query(self, board: Board, max_syzygy_size: int = 6) -> bool:
                 return False
         return True
 
+    def _categorize_dtz(self, dtz: int) -> "EndgameTablebase.DTZCategory":
+        """
+        Categorizes the Distance to Zeroing (DTZ) value into different categories.
+
+        :param dtz: The DTZ value to categorize, from our perspective.
+        :type dtz: int
+        :return: The category of the DTZ value.
+        :rtype: EndgameTablebase.DTZCategory
+        """
+        assert isinstance(
+            dtz, int
+        ), f"Expected DTZ to be of type int but got {type(dtz).__name__}."
+        if dtz < -100:
+            return EndgameTablebase.DTZCategory.BLESSED_LOSS
+        elif -100 <= dtz <= -1:
+            return EndgameTablebase.DTZCategory.UNCONDITIONAL_LOSS
+        elif dtz == 0:
+            return EndgameTablebase.DTZCategory.UNCONDITIONAL_DRAW
+        elif 1 <= dtz <= 100:
+            return EndgameTablebase.DTZCategory.UNCONDITIONAL_WIN
+        else:
+            return EndgameTablebase.DTZCategory.CURSED_WIN
+
+    def _compare_dtz(self, dtz: int, best_dtz: int, category: DTZCategory) -> int:
+        """
+        Compares two Distance to Zeroing (DTZ) values based on their categories.
+        Returns the best resulting DTZ value, based on the condition of the category.
+        For example, for the BLESSED_LOSS category, we want to save our loss as quickly as possible.
+        BLESSED_LOSS values are negative, thus we want to pick the biggest one.
+
+        :param dtz: The DTZ value to compare.
+        :type dtz: int
+        :param best_dtz: The best DTZ value so far.
+        :type best_dtz: int
+        :param category: The category of the DTZ value.
+        :type category: EndgameTablebase.DTZCategory
+        :return: The best resulting DTZ value, based on condition of the category.
+        :rtype: int
+        """
+        # We want to save our blessed loss as quickly as possible
+        if category == EndgameTablebase.DTZCategory.BLESSED_LOSS:
+            return max(dtz, best_dtz)
```

Comment:
This 50 move rule thing is really confusing me...I can't say I fully understood it. 

Won't block this for now, will discuss with you offline.